tools/autograd/templates/ManualFunctions.h

```diff
@@ -0,0 +1,29 @@
+// NB: Must be at the top of file to avoid including the deprecated "math.h".
+// https://stackoverflow.com/questions/6563810/m-pi-works-with-math-h-but-not-with-cmath-in-visual-studio
+#ifdef _MSC_VER
+#ifndef _USE_MATH_DEFINES
+#define _USE_MATH_DEFINES
+#endif
+#include <cmath>
+#endif
+
+#include "torch/csrc/autograd/generated/Functions.h"
+#include <ATen/Utils.h>
+#include <c10/core/TensorOptions.h>
+#include <ATen/WrapDimUtils.h>
+#include <ATen/WrapDimUtilsMulti.h>
+#include <ATen/SparseTensorUtils.h>
+#include <ATen/ExpandUtils.h>
+#include <ATen/core/Reduction.h>
+#include <ATen/Dispatch.h>
+#include <ATen/ScalarOps.h>
+
+#include <ciso646>
+#include <algorithm>
+#include <numeric>
+#include <functional>
+
+using at::Tensor;
+using at::Scalar;
+using at::IntArrayRef;
+using at::TensorList;
```

Comment:
Is ManualFunctions.h included in another C++ file in this stack of PRs, or is it just used in VariableType.cpp?

The `using at::Tensor;` declarations make Tensor, Scalar, IntArrayRef, and TensorList available to the global namespace. This could be bad if ManualFunctions.h ultimately gets included into e.g. torch.h, so it's usually better for `using blah` to appear in the .cpp files

Reply:
No it is not added anywhere else in this stack.
It's not actually a header file as it's just added to the cpp files.

Do you think I should remove the `.h` that can be confusing?